gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -80,29 +84,43 @@ def run(
 
         # convert translation direction in global frame using rotations.
         w_i2Ui1_measurements = BinaryMeasurementsUnit3()
+        w_i2Ui1_dict = {}
         for (i1, i2), i2Ui1 in i2Ui1_dict.items():
             if i2Ui1 is not None and wRi_list[i2] is not None:
+                measured = Unit3(wRi_list[i2].rotate(i2Ui1.point3()))
                 w_i2Ui1_measurements.append(
                     BinaryMeasurementUnit3(i2, i1, Unit3(wRi_list[i2].rotate(i2Ui1.point3())), noise_model)
                 )
+                w_i2Ui1_dict[(i1, i2)] = measured
 
         # sample indices to be used as projection directions
-        num_valid_measurements = len(w_i2Ui1_measurements)
-        indices = np.random.choice(
-            num_valid_measurements,
-            min(self._max_1dsfm_projection_directions, num_valid_measurements),
-            replace=False,
-        )
-
-        projection_directions = [w_i2Ui1_measurements[idx].measured() for idx in indices]
+        w_i2Ui1_list = [w_i2Ui1.measured().point3() for w_i2Ui1 in w_i2Ui1_measurements]
+        projection_directions = _sample_kde_directions(w_i2Ui1_measurements, self._max_1dsfm_projection_directions)
 
         # compute outlier weights using MFAS
         outlier_weights = []
+        ow_weights_dict = {}
 
         # TODO(ayush): parallelize this step.
         for direction in projection_directions:
             algorithm = MFAS(w_i2Ui1_measurements, direction)
-            outlier_weights.append(algorithm.computeOutlierWeights())
+            this_ow_dict = algorithm.computeOutlierWeights()
+            outlier_weights.append(this_ow_dict)
+            for (i2, i1), weight in this_ow_dict.items():
+                if (i1, i2) in ow_weights_dict:
+                    ow_weights_dict[(i1, i2)].append(weight)
+                else:
+                    ow_weights_dict[(i1, i2)] = [weight]
+
+        with open("result_metrics/proj_weights_1dsfm.txt", "w") as f:
+            f.write("# i1,i2,ux,uy,uz,weights_list\n")
+            for (i1, i2), weights_list in ow_weights_dict.items():
```

Comment:
Are we checking in this I/O code or is it for debugging?

Reply:
No this is all just debugging code. As mentioned in the description, PR is WIP, I will update it once the benchmarks are available. Thanks for taking a look!